docs/dev/datastream_execution_mode.md

```diff
@@ -0,0 +1,246 @@
+---
+title: "Execution Mode (Batch/Streaming)"
```

Comment:
Can you add "Execution Mode" to the glossary? 

Also, does it make sense to add a note to the `dev/batch/index.md` directing new applications to use bounded datastream or table? Or are we not there yet?

Reply:
I pushed a commit that adds it and I'm also recommending to use the DataStream API in the DataSet docs. I think it's time for a bold move... 😅